src/libs/actions/Report.js

```diff
@@ -2019,6 +2028,36 @@ function getCurrentUserAccountID() {
     return currentUserAccountID;
 }
 
+/**
+ * Update recently accessed reportID
+ *
+ * @param {String} currentReportID
+ * @param {Boolean} shouldRemoveCurrentReportID
+ * @returns {String} Returns the most recent accessed reportID
+ */
+function updateRecentlyAccessedReportID(currentReportID = '', shouldRemoveCurrentReportID = false) {
+    let reportIDs = _.filter(recentlyAccessedReportIDs, (reportID) => reportID !== currentReportID);
+    if (!_.isEmpty(currentReportID) && !shouldRemoveCurrentReportID) {
+        reportIDs.push(currentReportID);
+    }
+
+    // We only trace a limited number of recently accessed reportIDs
+    reportIDs = reportIDs.slice(-CONST.REPORT.MAX_RECENTLY_ACCESSED_REPORT_IDS_STACK_LIMIT);
+
+    // We want to save recently accessed reportIDs to Onyx to be used after refreshing the page
+    Onyx.set(ONYXKEYS.RECENTLY_ACCESSED_REPORT_ID_LIST, reportIDs);
+    recentlyAccessedReportIDs = reportIDs;
+
+    return reportIDs.length > 0 ? reportIDs[reportIDs.length - 1] : '';
+}
+
+/**
+ * @returns {String} Returns the most recent accessed reportID
+ */
+function getMostRecentlyAccessedReportID() {
+    return recentlyAccessedReportIDs.length > 0 ? recentlyAccessedReportIDs[recentlyAccessedReportIDs.length - 1] : '';
```

Comment:
I think it would be more cleaner if we return all report ids and use the last one on the place where we want to use it.

```js
updateRecentlyAccessedReportID(reportID, true);
const recentlyAccessedReportIDs = getMostRecentlyAccessedReportIDs();
const recentlyAccessedReportID = _.last(recentlyAccessedReportIDs);
```

and use it like that instead of returning a value from `updateRecentlyAccessedReportID`, what do you think?

Reply:
Hmm, I'd prefer not returning the reportID list because we don't want the reportID array to be potentially modified by client. 

> and use it like that instead of returning a value from `updateRecentlyAccessedReportID`, what do you think?

I think let method `updateRecentlyAccessedReportID` update the stack and return most recently accessed reportID is clean like the use case in method `leaveRoom`

Reply:
> I'd prefer not returning the reportID list because we don't want the reportID array to be potentially modified by client.

I am not sure i understood this, could you explain it further?

This kind of feels wrong to me returning the same thing from two functions making it duplicate and very limited, i think [this](https://github.com/Expensify/App/pull/31097#discussion_r1393918407) will be more cleaner. what do you think @techievivek

Reply:
Hmm, I think the current two methods are clean enough. See below 3 use cases
1. In method `leaveRoom`, we make a single method call
```js
const lastAccessedReportID = updateRecentlyAccessedReportID(reportID, true);
```
to remove the reportID from stack and return the most recently accessed reportID.
2. In the event listener, we call 
```js
Report.updateRecentlyAccessedReportID(reportID);
```
to update the reportID in the stack and omit the return value.
3. In `ReportScreen`, we don't need to update reportID in the stack. So we just call method
```js
const lastAccessedReportID = Report.getMostRecentlyAccessedReportID();
```
to read the most recently accessed reportID.

Reply:
Still, it doesn't make sense to me to have two separate functions returning the same value, and i don'
t expect to have a value to be returned from `updateRecentlyAccessedReportID`.

1. Update the recently accessed report id stack and get the value

```js
updateRecentlyAccessedReportID(reportID, true);
const recentlyAccessedReportIDs = getMostRecentlyAccessedReportIDs();
const recentlyAccessedReportID = _.last(recentlyAccessedReportIDs);
```

2. Get recently accessed report id without updating the value

```js
const recentlyAccessedReportIDs = getMostRecentlyAccessedReportIDs();
const recentlyAccessedReportID = _.last(recentlyAccessedReportIDs);
```

I'll let @techievivek weigh https://github.com/Expensify/App/pull/31097#discussion_r1393918407

Reply:
I got your point. I'm also fine to 
1. let method `updateRecentlyAccessedReportID` not return value
2. let method `getMostRecentlyAccessedReportID` to return most recently accessed reportID. I think it makes more sense the hide the logic the extract the reportID inside the method. 

What do you think?